timbermill-client/src/main/java/com/datorama/timbermill/ElasticsearchClient.java

```diff
@@ -124,83 +130,103 @@ private void deleteOldIndex(String currIndex) throws IOException {
         if (daysBackToDelete < 1){
             return;
         }
-        String indexPrefix = currIndex.split("-")[0];
-        String oldIndex = getTaskIndexWithEnv(indexPrefix, ZonedDateTime.now().minusDays(daysBackToDelete));
-        GetIndexRequest existsRequest = new GetIndexRequest(oldIndex);
+        String[] split = currIndex.split("-");
+        String indexPrefix = split[0];
+        String env = split[1];
+        String oldIndex = getTaskIndexWithEnv(indexPrefix, env, ZonedDateTime.now().minusDays(daysBackToDelete));
+        GetIndexRequest existsRequest = new GetIndexRequest().indices(oldIndex);
         boolean exists = client.indices().exists(existsRequest, RequestOptions.DEFAULT);
         if (exists) {
             DeleteIndexRequest request = new DeleteIndexRequest(oldIndex);
             client.indices().delete(request, RequestOptions.DEFAULT);
         }
     }
 
-    void close(){
+    void indexMetaDataEvents(String env, String... metadataEvents) {
+        String metadataIndex = getTaskIndexWithEnv(TIMBERMILL_INDEX_METADATA_PREFIX, env, ZonedDateTime.now());
+
+        BulkRequest bulkRequest = new BulkRequest();
+        for (String metadataEvent : metadataEvents) {
+            IndexRequest indexRequest = new IndexRequest(metadataIndex, TYPE).source(metadataEvent, XContentType.JSON);
+            bulkRequest.add(indexRequest);
+        }
+        try {
+            deleteOldIndex(metadataIndex);
+            client.bulk(bulkRequest, RequestOptions.DEFAULT);
+        } catch (IOException e){
+            LOG.error("Couldn't index metadata event with events " + Arrays.toString(metadataEvents) + " to elasticsearch cluster.", e);
+        }
+    }
+
+    public void close(){
         try {
             client.close();
         } catch (IOException e) {
             throw new ElasticsearchException(e);
         }
     }
 
-    private void bulkIndexByBulkSize(List<Event> events, int bulkSize, int tryNum) {
-        BulkRequest request = new BulkRequest();
-        int currBatch = 0;
-        int i = 0;
-        String index = getTaskIndexWithEnv(TIMBERMILL_INDEX_PREFIX, ZonedDateTime.now());
+    private void sendBulkRequest(BulkRequest request){
         try {
-            deleteOldIndex(index);
+            BulkResponse responses = client.bulk(request, RequestOptions.DEFAULT);
+            if (responses.hasFailures()) {
+                LOG.error("Couldn't bulk index tasks to elasticsearch cluster. Error: {}", responses.buildFailureMessage());
+            }
         } catch (IOException e) {
-            LOG.error("Could not delete index " + index, e);
+            LOG.error("Couldn't bulk index tasks to elasticsearch cluster.", e);
         }
-        //TODO not correct - Should be changed to Rollover
 
+    }
+
+    void index(Map<String, Task> tasksMap) {
+        BulkRequest request = new BulkRequest();
```

Comment:
naming of all different request variables is a little confusing, lets differntiate them e.g. bulkRequest